lib/Elastica/Aggregation/Filters.php

```diff
@@ -22,10 +22,12 @@ class Filters extends AbstractAggregation
      */
     public function addFilter(AbstractFilter $filter, $name = '')
     {
-        if (empty($name)) {
-            $filterArray[] = $filter->toArray();
+        $filterArray = array();
+
+        if (is_string($name)) {
```

Comment:
@stof Agree. There is something wrong here. Did you spot this because of the discussion or did you run already in some issues with it?